### PR TITLE
age fun for postgresql

### PR DESCRIPTION
**Pull request by xiaoyumoyu**
## Which issue does this PR close?
none

Closes #.

## Rationale for this change
implement age function for postgresql

## What changes are included in this PR?
add age function

## Are these changes tested?
yes

## Are there any user-facing changes?
no

### CODE REVIEW

File: extension/functions/src/postgres.rs [hunk 1]
```diff
@@ -14,3 +14,149 @@
 // KIND, either express or implied.  See the License for the
 // specific language governing permissions and limitations
 // under the License.
+
+use arrow::array::*;
+// use arrow::array::{ArrayRef, BooleanArray, Int64Array,StringBuilder,Float64Array};
```
Comment (wizardxz) [inline extension/functions/src/postgres.rs]: rm

File: extension/functions/src/postgres.rs [hunk 2]
```diff
@@ -14,3 +14,149 @@
...
+
+impl ScalarFunctionDef for AgeFunction {
+    fn name(&self) -> &str {
+        "age"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::exact(
+            vec![
+                DataType::Timestamp(TimeUnit::Nanosecond, None),
```
Comment (wizardxz) [inline extension/functions/src/postgres.rs:47]: could also be second/millisecond/microsecond

**Review by wizardxz — CHANGES_REQUESTED** [review]

File: extension/functions/src/postgres.rs (hunk 2 above)
Comment (wizardxz) [reply extension/functions/src/postgres.rs:47]: As long as test works, I think it is fine to only have nanosecond cc @sadboy

Comment (sadboy) [reply extension/functions/src/postgres.rs:47]: This should match the default timeunit of Postgres right? (I haven't checked what that is)

File: extension/functions/src/postgres.rs [hunk 3]
```diff
@@ -14,3 +14,143 @@
...
+use datafusion::logical_expr::Volatility;
+
+use datafusion_common::DataFusionError;
+use datafusion_expr::{
+    ReturnTypeFunction, ScalarFunctionDef, ScalarFunctionPackage, Signature,
+};
+use std::sync::Arc;
+
+#[derive(Debug)]
+pub struct AgeFunction;
```
Comment (sadboy) [inline extension/functions/src/postgres.rs:37]: Could you add a link to the function in Postgres doc? That makes it easier to cross reference the source of truth. 
e.g.
```suggestion
/// Postgres `age` function ([link to doc])
pub struct AgeFunction;
```

**Review by wizardxz — APPROVED** [review]

File: extension/functions/src/postgres.rs (hunk 2 above)
Comment (wizardxz) [reply extension/functions/src/postgres.rs:47]: The default is the microsecond. But looks like DF has type coercion from any timeunit to Nanosecond. So I think it is fine after I look at the tests

File: extension/functions/src/postgres.rs (hunk 3 above)
Comment (wizardxz) [reply extension/functions/src/postgres.rs:37]: Great point! cc @xiaoyumoyu

Comment (xiaoyumoyu) [reply extension/functions/src/postgres.rs:37]: ok！Should I submit it again right away after making the additions？

**Pull request merged**